bin/installer.js

```diff
@@ -61,76 +62,84 @@ const args = Object.assign(
   parseArgs(process.argv.slice(2), supportedArgs)
 );
 
-// Auto-detect if scss, stylus or react are installed in the project
-const projectHasStylus = () => {
-  return (
-    projectDeps.stylus ||
-    (projectDeps.hxmgulp && fs.existsSync(projectPath + 'node_modules/stylus'))
-  );
-};
+// auto-detect options based on already installed deps
+{
+  if (args.scss == null && projectDeps.sass) {
+    console.info('scss detected.');
+    args.scss = true;
+  }
+  const projectHasReact = () =>
+    projectDeps.react || projectDeps.preact || projectDeps.inferno;
+  if (args.react == null && projectHasReact()) {
+    const reactLike = projectDeps.react
+      ? 'React'
+      : projectDeps.preact
+      ? 'Preact'
+      : 'Inferno';
+    console.info(reactLike + ' detected.');
+    args.react = true;
+  }
 
-if (args.stylus == null && projectHasStylus()) {
-  console.info('Stylus detected.');
-  args.stylus = true;
+  const projectHasTypeScript = () => projectDeps.typescript;
+  if (args.typescript == null && projectHasTypeScript()) {
+    console.info('TypeScript detected.');
+    args.typescript = true;
+  }
 }
 
-if (args.scss == null && projectDeps.scss) {
-  console.info('scss detected.');
-  args.scss = true;
-}
+// ---------------------------------------------------------------------------
 
-const projectHasReact = () => {
-  return projectDeps.react || projectDeps.preact || projectDeps.inferno;
+const installDeps = {
+  ...hxmstylePkg.peerDependencies,
 };
-
-if (args.react == null && projectHasReact()) {
-  const reactLike = projectDeps.react
-    ? 'React'
-    : projectDeps.preact
-    ? 'Preact'
-    : 'Inferno';
-  console.info(reactLike + ' detected.');
-  args.react = true;
-}
-
-const projectHasTypeScript = () => projectDeps.typescript;
-
-if (args.typescript == null && projectHasTypeScript()) {
-  console.info('TypeScript detected.');
-  args.typescript = true;
-}
-
-const installDeps = hxmstylePkg.peerDependencies;
+Object.entries(hxmstylePkg.peerDependenciesMeta).forEach(([key, meta]) => {
+  // Skip installing "optional" peerDependences, unless they're
+  // explicitly flagged in `args` AND missing from projectPkg
+  if (
+    meta.optional &&
+    (!args[key] || projectDeps[key] || projectPkg.peerDependencies[key])
+  ) {
+    delete installDeps[key];
+  }
+});
+// Keeping track of the names of dependences that are installed by hxmstyle
+// (Currently these are only deps related to `args.scss`.)
```

Comment:
eslint and prettier always get installed but are not being added to dependenciesAdded, is that intentional?